### PR TITLE
Script to automate DevTools tasks

### PR DESCRIPTION
**Pull request by jecfish**
Automate various DevTools tasks:

1. Generate What's New in DevTools blog post templates in 7 languages
2. Generate What's New in DevTools blog post banner images in 7 languages
3. Generate What's New in DevTools blog outlines in 7 languages

... and more. Refer to `README.md` for instructions and all scripts.

### CODE REVIEW

**Review by matthiasrohmer — CHANGES_REQUESTED** [review]
That'll make things quicker to handle, nice! Some thoughts and questions: currently, `gulp-tasks` only holds tasks that are necessary to actually build the site, tools that help doing so live in `tools`. Can you please move devtools.js there, into a new folder? 

Ideally the template files `new-in-devtools.md` and `new-in-devtools-banner.svg` then also live alongside in the same directory. Because similar as with the gulp tasks, the template files you created are not consumed directly by 11ty and therefore it would be nice to separate them from it 🙂 

Also a more descriptive name for the tool would be great! Something like devtools-blog-bootstrap maybe? Anything more descriptive would help future maintainers I think.

**Comment by jecfish** [comment]
Thanks @matthiasrohmer. I have rename and organised the files. PTAL. It's a multi step automation process. I split them into different scripts.

File: .gitignore
Comment (matthiasrohmer) [inline .gitignore:15]: Similar to the earlier comments: I think every directory or file at root-level should have some relation to the actual 11ty setup, so I'd move this to tools/devtools/temp. This will also make it obvious in the .gitignore what it's for and won't draw confusion when working on things like https://github.com/GoogleChrome/developer.chrome.com/issues/65

File: package.json
Comment (matthiasrohmer) [inline package.json:74]: I try to pin all newly introduced dependencies to have all of them pinned at one point. The reasoning behind this is that having pinned dependencies usually solves a lot of errors with lock files for not so regular contributors.

File: tools/devtools/templates/new-in-devtools.md
Comment (matthiasrohmer) [inline tools/devtools/templates/new-in-devtools.md:1]: Sorry, didn't think about it earlier, also just an idea: as this is essentially a template you could also use any of the template languages that 11ty ships with. Nunjucks for example. Pro: you would not have to deal with RegEx patterns and String.replace, Con: you would need to wrap things like `{% include %}` with `{% raw %}`. Your call 🙂

File: tools/devtools/utils.mjs
Comment (matthiasrohmer) [inline tools/devtools/utils.mjs:1]: At a glance all the methods in your tool that make file operations are `async` already so I'd vote to use the variants from `fs/promises`. That usually makes things easier to refactor later on as well and saves from probably unexpected halts

File: tools/devtools/utils.mjs
Comment (matthiasrohmer) [inline tools/devtools/utils.mjs:71]: It's a shame DateTimeFormat does not allow custom formats, but what you could do is:

`return new Intl.DateTimeFormat("fr-CA", {year: 'numeric', month: '2-digit', day: '2-digit'}).format(Date.now());`

That's a little hacky though as the `fr-CA` locale is what takes care of the order then.

File: tools/devtools/utils.mjs
Comment (matthiasrohmer) [inline tools/devtools/utils.mjs:228]: Nit: `Appended`, just so it's clear it already happened

File: tools/devtools/wndt-bootstrap.mjs
Comment (matthiasrohmer) [inline tools/devtools/wndt-bootstrap.mjs:1]: Not too sure, but you'll probably want to move this after the `config()` call to ensure things are really setup before doing anything else

File: tools/devtools/wndt-bootstrap.mjs
Comment (matthiasrohmer) [inline tools/devtools/wndt-bootstrap.mjs:11]: Technically there would be `const {defaultLocale} = require('site/_data/site.json');` but since you need to hardcode locales in other places I think we shouldn't bother.

File: tools/devtools/wndt-outline.mjs
Comment (matthiasrohmer) [inline tools/devtools/wndt-outline.mjs:1]: Same as above, this should probably come after `config()`

File: tools/devtools/wndt-translation-bootstrap.mjs
Comment (matthiasrohmer) [inline tools/devtools/wndt-translation-bootstrap.mjs:32]: Why not use `languages` here as well?

**Review by matthiasrohmer — COMMENTED** [review]
Left some comments, let me know what you think!

File: .remarkignore
Comment (jecfish) [inline .remarkignore:1]: @matthiasrohmer I took your advise to use 11ty/nunjucks syntax. However, I got a warning from the `lint:md` npm task. 11ty syntax is not allowed (see `no-unescaped-template-tags.js`).  

I checked the `remark-cli` docs, I couldn't find a way to disable the warning for just a line (e.g. `eslint-disable-next-line`). The documentation suggests me to ignore the files in `.remarkignore`. 

At this point, I can either fallback to the previous regex version or ignore this file. Your call.

File: tools/devtools/utils.mjs
Comment (jecfish) [inline tools/devtools/utils.mjs:2]: Will remove this line if we add `nunjucks` as direct dependecy.

File: tools/devtools/utils.mjs
Comment (jecfish) [inline tools/devtools/utils.mjs:83]: Inspired by your previous review comment, I change most of the regex replace syntax with [nunjucks render](https://mozilla.github.io/nunjucks/api.html#render). 

I can't use 11ty syntax as I need to pre-render it beforehand.

File: tools/devtools/utils.mjs
Comment (jecfish) [inline tools/devtools/utils.mjs:9]: If you are wondering why I change the block syntax, it's because I don't want nunjucks to render `{% include }`, 11ty should handle it.

Using `{% raw}` to avoid rendering doesn't work in my case because I render the files twice. With `{% raw}`, the second render doesn't work.

**Comment by jecfish** [comment]
@matthiasrohmer, ready for review again!

**Comment by jecfish** [comment]
Hi @matthiasrohmer if you can have a look

**Comment by jecfish** [comment]
Ping @matthiasrohmer for review.

File: tools/devtools/utils.mjs
Comment (matthiasrohmer) [inline tools/devtools/utils.mjs:240]: I wonder if it's worth safe guarding this to prevent issues from getting created twice if the script is run a second time with the same variables by accident?

File: tools/devtools/utils.mjs
Comment (matthiasrohmer) [reply tools/devtools/utils.mjs:9]: Doesn't hurt much, as it's scoped to your tool anyway. Just for my own curiosity: the `{% raw %}` tags would be gone after the first render pass, no? Or am I missing something?

File: tools/devtools/utils.mjs
Comment (matthiasrohmer) [reply tools/devtools/utils.mjs:2]: Yes, let's add Nunjucks as a direct dependency.

File: package.json
Comment (matthiasrohmer) [reply package.json:100]: Yep pinning is good! npm would complain if there ever is a version mismatch. Also add nunjucks.

File: .remarkignore
Comment (matthiasrohmer) [reply .remarkignore:1]: Is this still valid? Now all templates seem to live in tools/devtools/templates. Or is remark complaining after generation?

File: tools/devtools/utils.mjs
Comment (jecfish) [reply tools/devtools/utils.mjs:240]: I have a safe guard when calling this function. The CLI will explicitly ask if we want to generate, see `wndt-translation-bootstrap` - `readline.question('Do you want to generate GitHub issues? y/[n]:` 

The worst case is we have duplicated issues that we need to manually clean up.

Comment (matthiasrohmer) [reply tools/devtools/utils.mjs:240]: Yep, I have seen this one 🙂 I just can imagine myself seeing pressing `y` in a rush - that's where I was coming from. But maybe even calling out the configured version in the prompt again would save me already.

File: tools/devtools/templates/new-in-devtools-image-snippet.md
Comment (matthiasrohmer) [inline tools/devtools/templates/new-in-devtools-image-snippet.md:2]: I was just trying this out and wondered where to upload them?

**Comment by matthiasrohmer** [comment]
Again, sorry for the long wait. Had another look and fiddled around with it. I don't quite get the idea of the image flow yet, but the others look fine to me 🙂

File: tools/devtools/templates/new-in-devtools-image-snippet.md
Comment (jecfish) [reply tools/devtools/templates/new-in-devtools-image-snippet.md:2]: https://web-dev-uploads.web.app/uploader

Comment (jecfish) [reply tools/devtools/templates/new-in-devtools-image-snippet.md:2]: It would be nice if I can just upload it using firebase api and get the url. But I don't have capacity to dig into that.

File: tools/devtools/utils.mjs
Comment (jecfish) [reply tools/devtools/utils.mjs:8]: yes, it will go after the first pass. I need two pass.

File: tools/devtools/utils.mjs
Comment (jecfish) [reply tools/devtools/utils.mjs:239]: Add version number and force it to be capital Y for additional safeguard.

**Review by jecfish — COMMENTED** [review]
Resolve all the comments. Please take a look!

**Review by matthiasrohmer — APPROVED** [review]
Nice! LGTM.

**Comment by matthiasrohmer** [comment]
Resolved the dependency issues. Needs content team approval, because Markdown files have been changed.

**Review by rachelandrew — APPROVED** [review]

**Pull request merged**
